drivers/i2s/i2s_cavs.c

```diff
@@ -79,15 +79,16 @@ struct stream {
 	struct ring_buf mem_block_queue;
 	void *mem_block;
 	bool last_block;
-	int (*stream_start)(struct stream *, struct i2s_cavs_ssp *const,
+	int (*stream_start)(struct stream *, struct i2s_cavs_ssp *volatile,
```

Comment:
What is the idea behind making this volatile instead of const? By const, i was ensuring the pointer variable doesn't get reassigned to anything else.

Reply:
Since the pointer is pointing to hardware registers, the idea is to force the compiler to read the register every time there is a read instead of re-using previously read values in the processor's machine registers.

Reply:
If that's the case, it should be something like "volatile struct i2s_cavs_ssp *const". No? Here the struct is volatile where as the pointer to the struct remains a constant.

Reply:
That's correct. I'll update it.